plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/store/handlers/ConnectorStoreEarlyStartupHandler.java

```diff
@@ -0,0 +1,109 @@
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.jetty.servlet.DefaultServlet;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.ui.IStartup;
+import org.osgi.framework.Bundle;
+import org.wso2.developerstudio.eclipse.esb.project.Activator;
+import org.wso2.developerstudio.eclipse.esb.project.servlets.ConnectorDownloadServlet;
+import org.wso2.developerstudio.eclipse.esb.project.servlets.ConnectorRetrievalServlet;
+import org.wso2.developerstudio.eclipse.esb.project.servlets.ConnectorSearchServlet;
+import org.wso2.developerstudio.eclipse.logging.core.IDeveloperStudioLog;
+import org.wso2.developerstudio.eclipse.logging.core.Logger;
+import org.wso2.developerstudio.eclipse.templates.dashboard.handlers.JettyServerHandler;
+
+/**
+ * Binds new servlets to paths of the jetty server
+ * 
+ */
+public class ConnectorStoreEarlyStartupHandler implements IStartup {
+
+    private static IDeveloperStudioLog log = Logger.getLog(Activator.PLUGIN_ID);
+    private static final String CONTEXT_PATH = "/project/connectors";
+    private static final String CONNECTOR_RETRIEVAL_SERVLET_PATH = "/list";
+    private static final String CONNECTOR_DOWNLOAD_SERVLET_PATH = "/download";
+    private static final String CONNECTOR_SEARCH_SERVLET_PATH = "/search";
+    private static final String WEB_APP_LOCATION = "HTMLPages";
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.eclipse.ui.IStartup#earlyStartup()
+     */
+    @Override
+    public void earlyStartup() {
+        JettyServerHandler jettyServerHandler = JettyServerHandler.getInstance();
+
+        // Registering servlet context handler of store connectors window
+        ServletContextHandler connectorsContext = new ServletContextHandler();
+
+        // Context path where static webpages are hosted
+        String webAppPath = "";
+        try {
+            // Get web app path from current bundle
+            webAppPath = getWebAppPath();
+        } catch (URISyntaxException uriException) {
+            log.error("Error resolving web app path", uriException);
+        } catch (IOException ioException) {
+            log.error("Error resolving web app path", ioException);
+        }
+        connectorsContext.setContextPath(CONTEXT_PATH);
+
+        // Adding Default servlet and Connector servlet
+        connectorsContext.addServlet(ConnectorRetrievalServlet.class, CONNECTOR_RETRIEVAL_SERVLET_PATH);
+        connectorsContext.addServlet(ConnectorDownloadServlet.class, CONNECTOR_DOWNLOAD_SERVLET_PATH);
+        connectorsContext.addServlet(ConnectorSearchServlet.class, CONNECTOR_SEARCH_SERVLET_PATH);
+        connectorsContext.addServlet(DefaultServlet.class, "/");
+        connectorsContext.setResourceBase(webAppPath);
+
+        // Registering the handler in the jetty server
+        jettyServerHandler.getHandlerCollection().addHandler(connectorsContext);
+        try {
+            connectorsContext.start();
+        } catch (Exception e) {
```

Comment:
Shall we catch the specific exception ?

Reply:
start() method in org.eclipse.jetty.util.component.LifeCycle throws Exception